flow/connectors/postgres/postgres.go

```diff
@@ -941,30 +941,48 @@ func (c *PostgresConnector) EnsurePullability(
 	return &protos.EnsurePullabilityBatchOutput{TableIdentifierMapping: tableIdentifierMapping}, nil
 }
 
-func (c *PostgresConnector) ExportSnapshot(ctx context.Context) (string, any, error) {
+func (c *PostgresConnector) ExportTxnSnapshot(ctx context.Context) (*protos.ExportTxnSnapshotOutput, any, error) {
 	var snapshotName string
 	tx, err := c.conn.Begin(ctx)
 	if err != nil {
-		return "", nil, err
+		return nil, nil, err
+	}
+	txNeedsRollback := true
+	defer func() {
+		if txNeedsRollback {
+			err := tx.Rollback(context.Background())
```

Comment:
```suggestion
			err := tx.Rollback(context.Background())
```
Should have at least a 5 minute timeout or something